tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+    {
+        function() return { x = 0  , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 32 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 64 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 96 , y = 0, width  = 32, height = 32, } end,
+        function() return { x = 128, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 160, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 192, y = 0, width  = 32, height = 32, } end,
+        function() return { x = 224, y = 0, width  = 32, height = 32, } end,
+    },
+
+    -- Corner case 4: A screen taller than more than 1 other screen.
+    -- this may cause some issues with client resize and drag&drop move
+    {
+        function()
+            return {
+                x = 0,
+                y = 0,
+                width  = math.floor(canvas_w/2),
+                height = canvas_h,
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 0,
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = math.floor(canvas_h/3),
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end,
+        function()
+            return {
+                x = math.floor(canvas_w/2)+1,
+                y = 2*math.floor(canvas_h/3),
+                width = math.floor(canvas_w/2),
+                height = math.floor(canvas_h/3),
+            }
+        end
+    },
+
+    -- The overlapping corner case isn't supported. There is valid use case,
+    -- such as a projector with a smaller resolution than the laptop screen
+    -- in non-scaling mirror mode, but it isn't worth the complexity it brings.
+}
+
+local function check_tag_indexes()
+    for s in screen do
+        for i, t in ipairs(s.tags) do
+            assert(t.index  == i)
+            assert(t.screen == s)
+        end
+    end
+end
+
+local colors = {
+    "#000030",
+    "#300000",
+    "#043000",
+    "#302E00",
+    "#002C30",
+    "#300030",
+    "#301C00",
+    "#140030",
+}
+
+-- Paint it black
+local function clear_screen()
+    for s in screen do
+        local sur = surface.widget_to_surface(
+            wibox.widget {
+                bg     = "#000000",
+                widget = wibox.container.background
+            },
+            s.geometry.width,
+            s.geometry.height
+        )
+        wallpaper.fit(sur, s, "#000000")
+    end
+end
+
+-- Make it easier to debug the tests by showing the screen geometry when the
+-- tests are executed.
+local function show_screens()
+    wallpaper.set(color("#000000")) -- Should this clear the wallpaper? It doesn't
```

Comment:
Nope, it's not worth reinventing the wheel. I forgot/missed the needed text part. Without the text, I'd say yes. ;-)